src/events/alb/HttpServer.js

```diff
@@ -0,0 +1,287 @@
+    if (hapiMethod === 'HEAD') {
+      log.notice(
+        'HEAD method event detected. Skipping HAPI server route mapping',
+      )
+
+      return
+    }
+
+    if (hapiMethod !== 'HEAD' && hapiMethod !== 'GET') {
+      // maxBytes: Increase request size from 1MB default limit to 10MB.
+      // Cf AWS API GW payload limits.
+      hapiOptions.payload = {
+        maxBytes: 1024 * 1024 * 10,
+        parse: false,
+      }
+    }
+
+    const hapiHandler = async (request, h) => {
+      this.#lastRequestOptions = {
+        headers: request.headers,
+        method: request.method,
+        payload: request.payload,
+        url: request.url.href,
+      }
+
+      // Payload processing
+      const encoding = detectEncoding(request)
+
+      request.payload = request.payload && request.payload.toString(encoding)
+      request.rawPayload = request.payload
+
+      // Incoming request message
+      log.notice()
+
+      log.notice()
+      log.notice(`${method} ${request.path} (λ: ${functionKey})`)
+
+      const response = h.response()
+
+      const event = new LambdaAlbRequestEvent(request).create()
+
+      const lambdaFunction = this.#lambda.get(functionKey)
+
+      lambdaFunction.setEvent(event)
+
+      let result
+      let err
+
+      try {
+        result = await lambdaFunction.runHandler()
+      } catch (_err) {
+        err = _err
+      }
+
+      log.debug('_____ HANDLER RESOLVED _____')
+
+      // Failure handling
+      let errorStatusCode = '502'
+      if (err) {
+        // Since the --useChildProcesses option loads the handler in
+        // a separate process and serverless-offline communicates with it
+        // over IPC, we are unable to catch JavaScript unhandledException errors
+        // when the handler code contains bad JavaScript. Instead, we "catch"
+        // it here and reply in the same way that we would have above when
+        // we lazy-load the non-IPC handler function.
+        if (this.#options.useChildProcesses && err.ipcException) {
+          return this.#reply502(
+            response,
+            `Error while loading ${functionKey}`,
+            err,
+          )
+        }
+
+        const errorMessage = (err.message || err).toString()
+
+        const re = /\[(\d{3})]/
+        const found = errorMessage.match(re)
+
+        if (found && found.length > 1) {
+          ;[, errorStatusCode] = found
+        } else {
+          errorStatusCode = '502'
+        }
+
+        // Mocks Lambda errors
+        result = {
+          errorMessage,
+          errorType: err.constructor.name,
+          stackTrace: this.#getArrayStackTrace(err.stack),
+        }
+
+        log.notice(`Failure: ${errorMessage}`)
+
+        if (!this.#options.hideStackTraces) {
+          log.error(err.stack)
+        }
+      }
+
+      let statusCode = 200
+      if (err) {
+        statusCode = errorStatusCode
+      }
+      response.statusCode = statusCode
```

Comment:
In this case shouldn't be better to have something like this?

```suggestion
      response.statusCode = result.statusCode
```

and it's also missing the headers from the response, so it could be something like this

```suggestion
      response.statusCode = result.statusCode
      response.headers = result.headers
```